qrexec-lib/unpack.c

```diff
@@ -154,7 +154,7 @@ static void fix_times_and_perms(const int fd,
     } else {
         /* Do not change the mode of what a symbolic link points to */
         if (!S_ISLNK(untrusted_hdr->mode) &&
-            fchmodat(fd, last_segment, untrusted_hdr->mode & 07777, 0))
+            fchmodat(fd, last_segment, untrusted_hdr->mode & 07777, AT_SYMLINK_NOFOLLOW))
```

Comment:
checking glibc git log says:
```
commit 65341f7bbea824d2ff9d37db15d8be162df42bd3
Author: Adhemerval Zanella <adhemerval.zanella@linaro.org>
Date:   Thu Sep 28 13:56:21 2023 -0300

    linux: Use fchmodat2 on fchmod for flags different than 0 (BZ 26401)
    
    Linux 6.6 (09da082b07bbae1c) added support for fchmodat2, which has
    similar semantics as fchmodat with an extra flag argument.  This
    allows fchmodat to implement AT_SYMLINK_NOFOLLOW and AT_EMPTY_PATH
    without the need for procfs.
    
    The syscall is registered on all architectures (with value of 452
    except on alpha which is 562, commit 78252deb023cf087).
    
    The tst-lchmod.c requires a small fix where fchmodat checks two
    contradictory assertions ('(st.st_mode & 0777) == 2' and
    '(st.st_mode & 0777) == 3').
```

Linux 6.6 is way too new to rely on this feature, and the fallback via procfs can't work here. Drop the change.

Reply:
Reimplemented as 075dfe462639f7d9a205138558cbf5a3a64d2366 using `fchmod()` and moved to a separate PR (#114).